src/components/WebexActivityStream/WebexActivityStream.js

```diff
@@ -138,19 +170,33 @@ Greeting.propTypes = {
   personName: PropTypes.string.isRequired,
 };
 
+export function TimeRuler({text}) {
+  return <ListSeparator className="time-ruler" role="listitem" text={text} />;
+}
+
+TimeRuler.propTypes = {
+  text: PropTypes.string.isRequired,
+};
+
 export default function WebexActivityStream(props) {
   const {roomID, adapters} = props;
   const {roomsAdapter, activitiesAdapter, peopleAdapter} = adapters;
   const {title, roomType} = useRoom(roomID, roomsAdapter);
-  const activityIDs = useActivityStream(roomID, roomsAdapter);
+  const activitiesData = useActivityStream(roomID, roomsAdapter);
   const personName = roomType === RoomType.DIRECT ? title : '';
-  const activities = activityIDs.map((activityID) => (
-    <WebexActivity key={activityID} activityID={activityID} adapters={{activitiesAdapter, peopleAdapter}} />
-  ));
+  const activities = activitiesData.map((activity) => {
+    if (activity.date) {
+      return <TimeRuler key={activity.date.toString()} text={formatTimeRulerText(activity.date)} />;
+    }
+
+    const activityID = activity;
```

Comment:
can't quite follow why we are declaring `activityID` again.

Reply:
It was for readability since `activity` could be an object or a string. I've removed it.